bindings/python/elliptics_session.cpp

```diff
@@ -2134,36 +2134,105 @@ void init_elliptics_session() {
+		     bp::args("key"),
+		     "read_json(key)\n"
+		     "    Read @key's json.\n"
+		     "    Return elliptics.AsyncResult.\n\n"
+		     "    async = session.read_json('key')\n"
+		     "    assert len(async.get()) == 1\n",
+		     "    result = async.get()[0]\n"
+		     "    print 'key has json: {}'.format(result.json)\n\n")
 
 		.def("read_data", &newapi::elliptics_session::read_data,
-		     (bp::arg("key"), bp::arg("offset") = 0, bp::arg("size") = 0))
+		     (bp::arg("key"), bp::arg("offset") = 0, bp::arg("size") = 0),
+		     "read_data(key, offset=0, size=0)\n"
+		     "    Read @key's data.\n"
+		     "    Return elliptics.AsyncResult.\n\n"
+		     "    async = session.read_data('key')\n"
+		     "    assert len(async.get()) == 1\n",
+		     "    result = async.get()[0]\n"
+		     "    print 'key has data: {}'.format(result.data)\n\n")
 
 		.def("read", &newapi::elliptics_session::read,
-		     (bp::arg("key"), bp::arg("offset") = 0, bp::arg("size") = 0))
+		     (bp::arg("key"), bp::arg("offset") = 0, bp::arg("size") = 0),
+		     "read(key, offset=0, size=0)\n"
+		     "    Read both @key's json and data.\n"
+		     "    Return elliptics.AsyncResult.\n\n"
+		     "    async = session.read('key')\n"
+		     "    assert len(async.get()) == 1\n",
+		     "    result = async.get()[0]\n"
+		     "    print 'key has json: {}, data: {}'.format(result.json, result.data)\n\n")
 
 		.def("write", &newapi::elliptics_session::write,
-		     (bp::arg("key"),
-		      bp::arg("json"), bp::arg("json_capacity"),
-		      bp::arg("data"), bp::arg("data_capacity")))
+		     bp::args("key", "json", "json_capacity", "data", "data_capacity"),
+		     "write(key, json, json_capacity, data, data_capacity)\n"
+		     "    Write @json and @data to @key.\n"
+		     "    -- json_capacity - size of reserved space for future updates of json\n"
+		     "       if json_capacity is 0, it will be set to size of @json\n"
+		     "    -- data_capacity - size of reserved space for future updates of data\n"
+		     "       if data_capacity is 0, it will be set to size of @data\n"
+		     "    Returns elliptics.AsyncResult.\n\n"
+		     "    async = session.write('key', '{\"some\":\"json\"}', 0, 'some data', 0)\n"
+		     "    for result in async:\n"
+		     "        print 'group: {}, status: {}, path: {}'.format(result.group_id,\n"
+		     "                                                       result.status, \n"
+		     "                                                       result.path")
 
 		.def("write_prepare", &newapi::elliptics_session::write_prepare,
-		     (bp::arg("key"),
-		      bp::arg("json"), bp::arg("json_capacity"),
-		      bp::arg("data"), bp::arg("data_offset"), bp::arg("data_capacity")))
+		     bp::args("key", "json", "json_capacity", "data", "data_offset", "data_capacity"),
+		     "write_prepare(key, json, json_capacity, data, data_offset, data_capacity)\n"
+		     "    Prepare place for @key and write json and/or first chunk of data.\n"
```

Comment:
I will change `place` to `space`, but it is not only allocation, so I leave `prepare` word.